src/main.js

```diff
@@ -6,6 +6,18 @@ var fs = require('fs');
 
 require('crash-reporter').start();
 
+// Load persisted state.
+var state = {};
+var statePath = path.join(app.getDataPath(), "state.json");
+try {
+  state = JSON.parse(fs.readFileSync(statePath, 'utf8'));
+} catch(e) {
+}
```

Comment:
That's because `fs.readFileSync` is a node module, which doesn't throw, but handles errors via a callback, which you haven't specified.  

As for `JSON.parse` it throws a type exception if the returned data isn't JSON.